src/ngraph/runtime/allocator.cpp

```diff
@@ -0,0 +1,52 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/allocator.hpp"
+
+ngraph::runtime::Allocator::~Allocator()
+{
+}
+
+class DefaultNgraphAllocator : public ngraph::runtime::Allocator
+{
+public:
+    void* Malloc(size_t size, size_t alignment)
+    {
+        void* ptr = ngraph::aligned_alloc(alignment, size);
```

Comment:
aligned_alloc is being removed in PR #2740 so you can't use it here. It does not support Windows and has some subtle use requirements. The problem here is that you need to maintain two separate pointers, one to the block of memory which was allocated and one to the aligned buffer within that block. You could return a shared_ptr since that can maintain two pointers internally for uses just like this.